gprofiler/metadata/application_identifiers.py

```diff
@@ -0,0 +1,248 @@
+import configparser
+import os.path
+import re
+from abc import ABCMeta, abstractmethod
+from typing import List, Optional
+
+from granulate_utils.linux.ns import resolve_host_path
+from psutil import NoSuchProcess, Process
+
+from gprofiler.log import get_logger_adapter
+
+_logger = get_logger_adapter(__name__)
+
+
+_PYTHON_BIN_RE = re.compile(r"^python([23](\.\d{1,2})?)?$")
+
+
+def _is_python_m_proc(process: Process) -> bool:
+    """
+    Checks whether the process ran as "python -m ..." pattern.
+    """
+    args = process.cmdline()
+
+    # Just to be on the safe side, module name must be run after the "-m" argument, python enforces it.
+    if len(args) < 3:
+        return False
+
+    return _is_python_bin(args[0]) and args[1] == "-m"
+
+
+def _is_python_bin(bin_name: str) -> bool:
+    return _PYTHON_BIN_RE.match(os.path.basename(bin_name)) is not None
+
+
+class _ApplicationIdentifier(metaclass=ABCMeta):
+    @abstractmethod
+    def get_application_name(self, process: Process) -> Optional[str]:
+        pass
+
+
+_IP_PORT_RE = re.compile(r"^\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}(?:\:\d{2,5})?$")  # Matches against (ip(:port))
+_NON_AVAILABLE_ARG = str()
+
+
+def _get_cli_arg_by_name(args: List[str], arg_name: str, check_for_equals_arg: bool = False) -> str:
+    if arg_name in args:
+        return args[args.index(arg_name) + 1]
+
+    if check_for_equals_arg:
+        for arg in args:
+            arg_key, _, arg_val = arg.rpartition("=")
+            if arg_key == arg_name:
+                return arg_val
+
+    return _NON_AVAILABLE_ARG
+
+
+def _get_cli_arg_by_index(args: List[str], index: int) -> str:
+    try:
+        return args[index]
+    except IndexError:
+        return _NON_AVAILABLE_ARG
+
+
+def _append_python_module_to_proc_wd(process: Process, module_path: str) -> str:
+    if os.path.isabs(module_path):
+        return module_path
+
+    return os.path.join(process.cwd(), module_path)
+
+
+class _GunicornApplicationIdentifier(_ApplicationIdentifier):
+    def get_application_name(self, process: Process) -> Optional[str]:
+        # As of gunicorn documentation the WSGI module name most probably will come from the cmdline and not from the
+        # config file / environment variables (they added the option to specify `wsgi_app`
+        # in the config file only in version 20.1.0)
+
+        if "gunicorn" not in _get_cli_arg_by_index(process.cmdline(), 0) or "gunicorn" not in _get_cli_arg_by_index(
```

Comment:
[Same](https://github.com/Granulate/gprofiler/pull/242/files#r777154006).
Also, what's the case of gunicorn `in` the second arg?

Reply:
@guybortnikov Forgot to answer my question

Reply:
gunicorn may be run directly or through python, for example:
a. `/usr/local/bin/gunicorn` 
b. `/usr/local/bin/python /usr/local/bin/gunicorn`

Reply:
@YishaiZinkin done?